src/path.jl

```diff
@@ -512,8 +512,8 @@ julia> mode(p"src/FilePathsBase.jl")
 -rw-r--r--
 ```
 """
-mode(fp::AbstractPath) = stat(fp).mode
-Base.filesize(fp::AbstractPath) = stat(fp).size
+mode(fp::AbstractPath) = Mode(stat(string(fp)).mode)
```

Comment:
`stat(string(fp))` isn't guaranteed to work for all `AbstractPath`s you probably only want this optimization for `SystemPath` in system.jl. I believe there are already shadow methods in there which probably should have been deleted a while ago. This would be a good case where `SystemPath` could have an internal optimization that doesn't change behaviour.